### DISCUSSION TITLE
How to Clear Chat Context/Memory in Open-WebUI?

### DISCUSSION THREAD

**Discussion by astkwcz** [open-webui/open-webui#7827] [Q&A] [answered]
Hi everyone,

I’ve noticed an issue while using Open-WebUI, and I’m wondering if there’s a way to address it.

**Problem:**

When I add documents to Open-WebUI and engage in a chat about a specific document (let’s call it Document X), the system seems to build context over the course of the conversation. However, if I then ask a question about a different document (Document Y) within the same chat, I often get incorrect responses. It feels like the context from Document X is affecting the answers about Document Y.

When I start a new chat and ask questions about Document Y, the responses are usually correct. This suggests that the context or memory from the ongoing chat influences the model’s responses.

**Question:**

Is there a way to clear or reset the context/memory for a specific chat so that previous prompts and discussions don’t affect the responses to new prompts? If not, is there a recommended workflow to avoid such issues when switching between different topics or documents?

Any insights or suggestions would be greatly appreciated. Thanks in advance for your help!Hi everyone,

I’ve noticed an issue while using Open-WebUI, and I’m wondering if there’s a way to address it.

**Comment by JZhouKai** [comment]
I have the same problem. When using the connected dify workflow dialogue, the system automatically builds the context, causing the parsing parameters of one of my workflow nodes to always be affected by the dialogue history. How can I solve this problem?

**Comment by richtong** [comment]
Great question. General question can you also cache context as well that is cache the kv store like l. Claude does.

**Comment by mattia-intellilab** [comment]
I have found a function that lets you define a number of recent messages that are being appended to the chat history. 
You may find inspiration in this function: [](https://openwebui.com/f/hub/context_clip_filter)

**Reply by astkwcz** [reply]
> I have found a function that lets you define a number of recent messages that are being appended to the chat history. You may find inspiration in this function:

@mattia-intellilab I believe you forgot to provide the function name. Could you please include it?

**Reply by mattia-intellilab** [reply, marked as answer]
Wow... sorry for that. 

You can find the function here: https://openwebui.com/f/hub/context_clip_filter

**Comment by richtong** [comment]
Oh thanks. Perfect!

Sent via Superhuman iOS ( ***@***.*** )
